lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -46,6 +67,18 @@ def need_two_factor_authentication?(request)
           true
         end
 
+        def otp_code
+          direct_otp
```

Comment:
If we want to maintain backwards compatibility, this should point to `totp_code` (i.e. `generate_totp_code` as you have it named). In the specs in this PR that test the direct OTP scenario, they would call `user.direct_otp`, not `user.otp_code`.